packages/site/src/pages/index.tsx

```diff
@@ -123,9 +125,38 @@ const Index = () => {
     }
   };
 
-  const handleSendHelloClick = async () => {
+  const handleGetPublicKey = async () => {
     try {
-      await sendHello();
+      const pk = await getPublicKey();
+      console.log(`Public key retrieved: ${pk}`);
+    } catch (e) {
+      console.error(e);
+      dispatch({ type: MetamaskActions.SetError, payload: e });
+    }
+  };
+
+  const handleSignMessageSecp256k1 = async () => {
+    try {
+      const message = Math.random().toString();
+      const encoder = new TextEncoder();
+      const byteArray = encoder.encode(message);
+
+      const signature = await signMessageSecp256k1(byteArray);
+      console.log(`Signature: ${signature}`);
+    } catch (e) {
+      console.error(e);
+      dispatch({ type: MetamaskActions.SetError, payload: e });
+    }
+  };
+
+  const handleSignMessageEd25519 = async () => {
+    try {
+      const message = Math.random().toString();
+      const encoder = new TextEncoder();
+      const byteArray = encoder.encode(message);
+
+      const signature = await signMessageEd25519(byteArray);
+      console.log(`Signature: ${signature}`);
     } catch (e) {
       console.error(e);
       dispatch({ type: MetamaskActions.SetError, payload: e });
```

Comment:
nit: These two functions are identical excpt for one line `const signature = await signMessageEd25519(byteArray);`. Maybe we should write a `handleSignMessage` helper which accepts the signing function as an argument?

Reply:
Reworked the test site frontend as a single react component.